integrations/pjrt/src/iree_pjrt/common/layout_utils.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cstdint>
+#include <vector>
+
+#include "iree/hal/buffer_view.h"
+#include "xla/pjrt/c/pjrt_c_api.h"
+
+namespace iree::pjrt {
+
+class ApiMemoryLayout {
+ public:
+  ApiMemoryLayout() = default;
+
+  void InitializeDenseRowMajorStrided(size_t rank, const int64_t *dims,
+                                      size_t unit_stride_bytes);
+  void InitializeDenseRowMajorTiled(int64_t rank);
```

Comment:
Is there only one tiled? (I feel like the above is unambiguous but for tiled I don't know what the default would be).

Reply:
I dunno. This is what I got by spelunking through a bunch of sparsely documented XLA internals. I think there is just one per rank unless tiled.